midi_app_controller/state/state_manager.py

```diff
@@ -195,3 +205,42 @@ def start_handling(self) -> None:
             midi_in=self._midi_in,
             midi_out=self._midi_out,
         )
+
+    def save_state(self):
+        AppState(
+            selected_controller_path=self.selected_controller.path
```

Comment:
Maybe if the if statement doesn't fit in one line, move it outside of the call. It would make it more readable in my opinion.